src/web/app/pages-instructor/instructor-session-individual-extension-page/instructor-session-individual-extension-page.component.ts

```diff
@@ -0,0 +1,503 @@
+  studentDeadlines: Record<string, number> = {};
+  instructorDeadlines: Record<string, number> = {};
+
+  isLoadingAllStudents: boolean = true;
+  hasLoadedAllStudentsFailed: boolean = false;
+  isLoadingAllInstructors: boolean = true;
+  hasLoadedAllInstructorsFailed: boolean = false;
+  isLoadingFeedbackSession: boolean = true;
+  hasLoadingFeedbackSessionFailed: boolean = false;
+  isSubmittingDeadlines: boolean = false;
+
+  ngOnInit(): void {
+    this.route.queryParams.subscribe((queryParams: any) => {
+      this.courseId = queryParams.courseid;
+      this.feedbackSessionName = queryParams.fsname;
+      this.loadFeedbackSessionAndIndividuals();
+    });
+  }
+
+  constructor(
+    private statusMessageService: StatusMessageService,
+    private feedbackSessionsService: FeedbackSessionsService,
+    private studentService: StudentService,
+    private ngbModal: NgbModal,
+    private route: ActivatedRoute,
+    private courseService: CourseService,
+    private tableComparatorService: TableComparatorService,
+    private instructorService: InstructorService,
+  ) { }
+
+  /**
+   * Loads a feedback session.
+   */
+  loadFeedbackSessionAndIndividuals(): void {
+    this.isLoadingAllStudents = true;
+    this.hasLoadedAllStudentsFailed = false;
+    this.isLoadingFeedbackSession = true;
+    this.hasLoadingFeedbackSessionFailed = false;
+    this.isLoadingAllInstructors = true;
+    this.hasLoadedAllStudentsFailed = false;
+    forkJoin([this.courseService.getCourseAsInstructor(this.courseId),
+      this.feedbackSessionsService.getFeedbackSession({
+        courseId: this.courseId,
+        feedbackSessionName: this.feedbackSessionName,
+        intent: Intent.FULL_DETAIL,
+      }),
+    ]).pipe(finalize(() => {
+      this.isLoadingFeedbackSession = false;
+    })).subscribe((value: any[]) => {
+      const course = value[0] as Course;
+      this.courseName = course.courseName;
+      const feedbackSession = value[1] as FeedbackSession;
+      this.setFeedbackSessionDetails(feedbackSession);
+      this.getAllStudentsOfCourse(); // Both students and instructors need feedback ending time.
+      this.getAllInstructorsOfCourse();
+    }, (resp: ErrorMessageOutput) => {
+      this.statusMessageService.showErrorToast(resp.error.message);
+      this.hasLoadingFeedbackSessionFailed = true;
+      this.isLoadingAllStudents = false;
+      this.isLoadingAllInstructors = false;
+    },
+    );
+  }
+
+  /**
+   * Gets all students of a course.
+   */
+  private getAllStudentsOfCourse(): void {
+    this.studentService
+      .getStudentsFromCourse({ courseId: this.courseId })
+      .pipe(finalize(() => { this.isLoadingAllStudents = false; }),
+        map((students: Students) => {
+          this.studentsOfCourse = students.students.map((student) => this.mapStudentToStudentModel(student));
+        }))
+      .subscribe(() => {
+        this.initialSortOfStudents();
+      },
```

Comment:
As far as possible, try to avoid side effects within pipe. If you have to perform operations with side effects, use `tap` instead of map. In this case, something like this would be suffice.

```suggestion
      .pipe(finalize(() => { this.isLoadingAllStudents = false; }),
        map((students: Students) => students.students.map((student) => this.mapStudentToStudentModel(student))),
      )
      .subscribe(
        (students: StudentExtensionTableColumnModel[]) => {
          this.studentsOfCourse = students;
          this.initialSortOfStudents();
        },
```

Reply:
The same applies for the corresponding method for instructors

Reply:
Makes sense, good catch, I didn't give the side effects portion much thought before